.github/workflows/main.yml

```diff
@@ -57,26 +68,25 @@ jobs:
       PYTHON: ${{ matrix.python-version }}
     steps:
       - uses: actions/checkout@v4
+
       - name: Ensure requests to production domains fail
         if: runner.os == 'Linux'
         run: |
           echo "203.0.113.0 logfire.dev" | sudo tee -a /etc/hosts
           echo "203.0.113.0 logfire-api.pydantic.dev" | sudo tee -a /etc/hosts
           echo "203.0.113.0 logfire.pydantic.dev" | sudo tee -a /etc/hosts
-      - name: set up python
-        uses: actions/setup-python@v5
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
         with:
-          python-version: ${{ matrix.python-version }}
-      - uses: eifinger/setup-rye@v2.6
-      - name: Pin python-version ${{ matrix.python-version }}
-        run: rye pin ${{ matrix.python-version }}
-      - run: rye config --set-bool behavior.use-uv=true
-      # Update all dependencies to the latest version possible
-      - run: rye sync --update-all
-      - run: |
-          pip install uv
-          uv pip install "logfire-api @ file://logfire-api"
-      - run: rye show
+          version: "0.4.12"
+          enable-cache: true
+
+      - name: Set up Python ${{ matrix.python-version }}
+        run: uv python install ${{ matrix.python-version }}
+
+      - run: uv sync --frozen
```

Comment:
This means we use the lockfile. We want to be always using the latest versions of things in CI tests, that's important.

Reply:
Why do other `uv sync` lines here have `--python 3.12` but not this one?